test/integration/integration_test.go

```diff
@@ -3,244 +3,106 @@
 }
 
 type TestConfig struct {
 	operatorNamespace string
 	clusterName       string
+	manifestsDir      string
 	kubeConfig        string
-
-	t *testing.T
 }
 
-func NewTestConfig(t *testing.T) *TestConfig {
-	config := &TestConfig{t: t}
-
+func NewTestConfig(t *testing.T, clusterName string, manifestsDir string) *TestConfig {
 	// Check prerequisites
 	kubeConfig := os.Getenv("KUBECONFIG")
 	if len(kubeConfig) == 0 {
 		t.Fatalf("KUBECONFIG is required")
 	}
 	// The operator-sdk uses KUBERNETES_CONFIG...
 	os.Setenv("KUBERNETES_CONFIG", kubeConfig)
-	config.kubeConfig = kubeConfig
 
-	clusterName := os.Getenv("CLUSTER_NAME")
 	if len(clusterName) == 0 {
-		t.Fatalf("CLUSTER_NAME is required")
+		t.Fatalf("cluster name is required")
+	}
+
+	if len(manifestsDir) == 0 {
+		t.Fatalf("manifests directory is required")
 	}
-	config.clusterName = clusterName
 
-	namespace, err := k8sutil.GetWatchNamespace()
-	if err != nil {
-		namespace = "default"
-		os.Setenv("WATCH_NAMESPACE", namespace)
+	return &TestConfig{
+		clusterName:       clusterName,
+		kubeConfig:        kubeConfig,
+		manifestsDir:      manifestsDir,
+		operatorNamespace: "openshift-cluster-ingress-operator",
 	}
-	config.operatorNamespace = namespace
+}
+
+func (tc *TestConfig) setup(t *testing.T) {
+	// uninstall the CVO
+	tc.runShellCmdNonFatal(t, `oc patch -n openshift-cluster-version daemonsets/cluster-version-operator --patch '{"spec": {"template": {"spec": {"nodeSelector": {"node-role.kubernetes.io/fake": ""}}}}}'`)
+
+	// uninstall tectonic-ingress
+	tc.runShellCmdNonFatal(t, `oc delete namespaces/openshift-ingress`)
+
+	// uninstall any existing operator
+	tc.uninstallOperator(t)
 
-	return config
+	// reinstall the operator
+	tc.runShellCmdNonFatal(t, fmt.Sprintf(`oc apply -f %s`, tc.manifestsDir))
 }
 
-func (tc *TestConfig) startOperator() {
-	resource := "ingress.openshift.io/v1alpha1"
-	kind := "ClusterIngress"
-	resyncPeriod := 10 * time.Minute
-	tc.t.Logf("Watching %s, %s, %s, %d", resource, kind, tc.operatorNamespace, resyncPeriod)
-	sdk.Watch(resource, kind, tc.operatorNamespace, resyncPeriod)
-	sdk.Handle(stub.NewHandler())
-	go sdk.Run(context.TODO())
+func (tc *TestConfig) uninstallOperator(t *testing.T) {
+	tc.runShellCmdNonFatal(t, `oc delete -n openshift-cluster-ingress-operator --force --grace-period=0 clusteringresses/default`)
+	tc.runShellCmdNonFatal(t, `oc delete namespaces/openshift-cluster-ingress-operator`)
+	tc.runShellCmdNonFatal(t, `oc delete namespaces/openshift-cluster-ingress-router`)
+	tc.runShellCmdNonFatal(t, `oc delete clusterroles/cluster-ingress-operator:operator`)
+	tc.runShellCmdNonFatal(t, `oc delete clusterroles/cluster-ingress:router`)
+	tc.runShellCmdNonFatal(t, `oc delete clusterrolebindings/cluster-ingress-operator:operator`)
+	tc.runShellCmdNonFatal(t, `oc delete clusterrolebindings/cluster-ingress:router`)
+	tc.runShellCmdNonFatal(t, `oc delete customresourcedefinition.apiextensions.k8s.io/clusteringresses.ingress.openshift.io`)
```

Comment:
Could you replace all the individual deletes with `tc.runShellCmdNonFatal(t, fmt.Sprintf(`oc delete -f %s -R`, tc.manifestsDir))`?

Reply:
Extracted all this into a script